pyatlan/model/assets/core/column.py

```diff
@@ -1574,7 +1574,7 @@ def create(
                 database_qualified_name=database_qualified_name,
             )
 
-            if parent_type == Table:
+            if parent_type in {Table, SnowflakeDynamicTable}:
                 column.table_qualified_name = parent_qualified_name
                 column.table = Table.ref_by_qualified_name(parent_qualified_name)
                 column.table_name = parent_name
```

Comment:
I have a doubt here. I noticed that the column has the `attribute snowflake_dynamic_table`. Shouldn't we use that here 🤔 

```suggestion
            elif parent_type == SnowflakeDynamicTable:
                column.table_qualified_name = parent_qualified_name
                column.snowflake_dynamic_table = SnowflakeDynamicTable.ref_by_qualified_name(
                    parent_qualified_name
                )
                column.table_name = parent_name
```

This is similar to what we do for other `parent_type` values, such as `MaterializedView`:

```py
     elif parent_type == MaterialisedView:
                column.view_qualified_name = parent_qualified_name
                column.materialised_view = MaterialisedView.ref_by_qualified_name(
                    parent_qualified_name
                )
                column.view_name = parent_name
  ```